php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1479 @@
+import static org.sonar.php.api.PHPKeyword.IMPLEMENTS;
+import static org.sonar.php.api.PHPKeyword.INSTANCEOF;
+import static org.sonar.php.api.PHPKeyword.INTERFACE;
+import static org.sonar.php.api.PHPKeyword.LIST;
+import static org.sonar.php.api.PHPKeyword.NEW;
+import static org.sonar.php.api.PHPKeyword.STATIC;
+import static org.sonar.php.api.PHPKeyword.TRAIT;
+import static org.sonar.php.api.PHPKeyword.USE;
+import static org.sonar.php.api.PHPKeyword.YIELD;
+import static org.sonar.php.api.PHPPunctuator.AMPERSAND;
+import static org.sonar.php.api.PHPPunctuator.ARROW;
+import static org.sonar.php.api.PHPPunctuator.COLON;
+import static org.sonar.php.api.PHPPunctuator.COMMA;
+import static org.sonar.php.api.PHPPunctuator.DEC;
+import static org.sonar.php.api.PHPPunctuator.DIV;
+import static org.sonar.php.api.PHPPunctuator.DOLLAR_LCURLY;
+import static org.sonar.php.api.PHPPunctuator.DOT;
+import static org.sonar.php.api.PHPPunctuator.DOUBLEARROW;
+import static org.sonar.php.api.PHPPunctuator.DOUBLECOLON;
+import static org.sonar.php.api.PHPPunctuator.ELIPSIS;
+import static org.sonar.php.api.PHPPunctuator.EQU;
+import static org.sonar.php.api.PHPPunctuator.EQUAL;
+import static org.sonar.php.api.PHPPunctuator.EQUAL2;
+import static org.sonar.php.api.PHPPunctuator.GE;
+import static org.sonar.php.api.PHPPunctuator.GT;
+import static org.sonar.php.api.PHPPunctuator.INC;
+import static org.sonar.php.api.PHPPunctuator.LBRACKET;
+import static org.sonar.php.api.PHPPunctuator.LCURLYBRACE;
+import static org.sonar.php.api.PHPPunctuator.LE;
+import static org.sonar.php.api.PHPPunctuator.LPARENTHESIS;
+import static org.sonar.php.api.PHPPunctuator.LT;
+import static org.sonar.php.api.PHPPunctuator.MINUS;
+import static org.sonar.php.api.PHPPunctuator.MOD;
+import static org.sonar.php.api.PHPPunctuator.NOTEQUAL;
+import static org.sonar.php.api.PHPPunctuator.NOTEQUAL2;
+import static org.sonar.php.api.PHPPunctuator.NOTEQUALBIS;
+import static org.sonar.php.api.PHPPunctuator.PLUS;
+import static org.sonar.php.api.PHPPunctuator.QUERY;
+import static org.sonar.php.api.PHPPunctuator.RBRACKET;
+import static org.sonar.php.api.PHPPunctuator.RCURLYBRACE;
+import static org.sonar.php.api.PHPPunctuator.RPARENTHESIS;
+import static org.sonar.php.api.PHPPunctuator.SL;
+import static org.sonar.php.api.PHPPunctuator.SR;
+import static org.sonar.php.api.PHPPunctuator.STAR;
+
+public class NewPHPGrammar {
+
+  private final GrammarBuilder<InternalSyntaxToken> b;
+  private final TreeFactory f;
+
+  public NewPHPGrammar(GrammarBuilder<InternalSyntaxToken> b, TreeFactory f) {
+    this.b = b;
+    this.f = f;
+  }
+
+  public CompilationUnitTree COMPILATION_UNIT() {
+    return b.<CompilationUnitTree>nonterminal(PHPLexicalGrammar.COMPILATION_UNIT).is(
+      f.compilationUnit(b.optional(SCRIPT()), b.optional(b.token(PHPLexicalGrammar.SPACING)), b.token(PHPLexicalGrammar.EOF)));
+  }
+
+  public ScriptTree SCRIPT() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)